File: Problem258.c
/*
Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.
*/


#include<stdio.h>

int addDigits(int num) 
{
    // return (num - 1) % 9 + 1;

    while(num >= 10)
    {
        int a = num / 10;
        int b = num % 10;
        num = a + b;
    }


    return num;
}

int main()
{
    printf("%d\n",addDigits(38));
    return 0;
}

File: Problem258.js
/*
Given an integer num, repeatedly add all its digits until the result has only one digit, and return it.
*/

/**
 * @param {number} num
 * @return {number}
 */
var addDigits = function (num) {
    let sum = 0;

    while (num > 9) {
        sum = 0;
        while (num != 0) {
            let digit = num % 10;
            sum += digit;
            num = Math.floor(num / 10);
        }
        num = sum;
    }
    return num;

};

console.log(addDigits(38));